src/containers/EventCreationPage.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useState } from "react";
+import { useHistory } from "react-router-dom";
+import "../styles/EventCreationPage.css";
+import Typography from "@material-ui/core/Typography";
+import Button from "@material-ui/core/Button";
+import CancelModal from "../components/EventCreationPage/CancelModal";
+import Map from "../components/EventCreationPage/Map";
+import NextButton from "../components/EventCreationPage/NextButton";
+import BackButton from "../components/EventCreationPage/BackButton";
+import FormField from "../components/EventCreationPage/FormField";
+import Stepper from "../components/EventCreationPage/Stepper";
+import SelectDateModal from "../components/EventCreationPage/SelectDateModal";
+import { useEventMutation } from "../graphql/mutations/hooks/events";
+import { useMutation } from "@apollo/react-hooks";
+import { ADD_EVENT } from "../graphql/mutations/templates/events";
+
+const EventCreationPage = () => {
+  const history = useHistory();
+  const [addEvent] = useMutation(ADD_EVENT);
+
+  const [openCancelModal, setOpenHandleModal] = useState(false);
+  const [openDateModal, setOpenDateModal] = useState(false);
+  const [complete, setComplete] = useState(false);
+
+  const [eventName, setEventName] = useState<string>("");
+  const [eventDate, setEventDate] = useState<Date | null>(null);
+  const [eventLocation, setEventLocation] = useState<string>("");
+
+  const [activeStep, setActiveStep] = useState<number>(0);
+
+  const handleOpenCancelModal = () => setOpenHandleModal(true);
+  const handleCloseCancelModal = () => setOpenHandleModal(false);
+
+  const handleOpenDateModal = () => setOpenDateModal(true);
+  const handleCloseDateModal = () => setOpenDateModal(false);
+
+  const handleNameChange = (e: any) => {
+    setEventName(e.target.value);
+  };
+  const handleDateChange = (e: any) => {
+    setEventDate(e.target.value);
+  };
+  const handleLocationChange = (e: any) => {
+    setEventLocation(e.target.value);
+  };
+
+  const handleNext = () => {
+    setActiveStep((prevStep) => prevStep + 1);
+  };
+  const handleBack = () => {
+    setActiveStep((prevStep) => prevStep - 1);
+  };
+
+  const dateParts: {
+    year?: string;
+    month?: string;
+    day?: string;
+    literal?: string;
+  } = eventDate
+      ? new Intl.DateTimeFormat().formatToParts(eventDate).reduce(
+        (obj, currentPart) => ({
+          ...obj,
+          [currentPart.type]: currentPart.value,
+        }),
+        {}
+      )
+<<<<<<< HEAD
+    : {};
+
+  const handleComplete = () => {
+    setComplete(true);
+    addEvent({
+      variables: {
+        name: eventName,
+        eventDate:
+          dateParts.year &&
+          dateParts.month &&
+          dateParts.day &&
+          `${dateParts.year}-${dateParts.month.padStart(
+            2,
+            "0"
+          )}-${dateParts.day.padStart(2, "0")}`,
+        createdBy: 1,
+        isActive: true,
+      },
+    });
+    history.replace("/events");
```

Comment:
Hi @wongcmichelle, is there a better way to take the user back to /events and re-render the page so the new event appears?

Reply:
What is the issue with the way you're doing it now (using `useHistory()`)?